parser.py

```diff
@@ -0,0 +1,118 @@
+class StrTableParser(object):
+    """
+    Parser of string representation tables
+
+    Example of basic usage:
+    >>> StrTableParser(table=f'| head |\\n| body_1 |\\n| body_2 |').container
+    {'head': ['body_1', 'body_2']}
+    """
+
+    def __init__(self, table=None):
+        self._container = {}
+        self._table = None
+        self._fields = None
+        self._body = None
+
+        if table:
+            self.table = table
+
+    @property
+    def table(self):
+        if self._table:
+            return self._table
+        else:
+            raise ValueError('Table is empty.')
```

Comment:
Does it make sense to move it to setter since it is a check only on empty value? I mean the cases when someone calls setter with empty value are rare. The reason it's here only that I wanted a graceful failure in case someone did next:
`StrTableParser().container`